node/src/test/scala/coop/rchain/finalization/FinalizationSpec.scala

```diff
@@ -126,6 +126,24 @@ class FinalizationSpec extends FlatSpec with Matchers {
           } yield r
       }
     }
+
+    def runInfinite: F[Unit] = {
+      val nets = List(10).map(genNet)
+
+      // This recursive function never finished
+      nets.tailRecM[F, Unit] { networks =>
+        val newNetState = networks.traverse {
+          case (net, name) =>
+            for {
+              net1_       <- runSections(net, List((1, .0f)), s"start-$name")
```

Comment:
I think here you need to add randomness to `runSections`, and also use `split` and `>|<`, to generate completely random DAGs.

Reply:
Done in [04532a1](https://github.com/rchain/rchain/pull/3601/commits/04532a1af56b635b391a67f6eb12add6214c6655).